### PR TITLE
Build with incompatible_disallow_empty_glob

### PR DESCRIPTION
**Pull request by limdor**
In order to flip the flag, all downstream projects should be adapted. However, it is hard to fix them all if there are constant regressions. Adding it to the CI will ensure that once the project can build with incompatible_disallow_empty_glob it can keep building like that.
See: bazelbuild/bazel#15327

### CODE REVIEW

**Comment by limdor** [comment]
@comius now is ready for review, it took me longer than expected to find the time to fix it. If you could take a look relatively soon it would be good to prevent regressions again.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ tasks:
       - rm -rf $HOME/bazeltest
       - mkdir $HOME/bazeltest
     build_flags:
+      - "--incompatible_disallow_empty_glob"
```
[the remark this answers is not in the document]
Comment (limdor) [reply .bazelci/presubmit.yml]: Done

**Review by meteorcloudy — APPROVED** [review]
Thanks!

**Comment by meteorcloudy** [comment]
We recently added the Bzlmod builds in the presubmit, unfortunately it's broken by this PR due to a different upb version:  https://buildkite.com/bazel/google-bazel-presubmit/builds/62549#0185a573-73f3-4f03-b5a6-4615047ae35b
```
(10:11:01) ERROR: Traceback (most recent call last):
	File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/upb~0.0.0-20220602-e5f2601/BUILD", line 718, column 16, in <toplevel>
		srcs = glob([
Error in glob: glob pattern 'upbc/**/*' didn't match anything, but allow_empty is set to False (the default value of allow_empty can be set with --incompatible_disallow_empty_glob).
```
Rebasing the PR to HEAD will reproduce this.

Ideally we should upgrade upb to a newer version (https://github.com/bazelbuild/bazel-central-registry/tree/main/modules/upb), if that doesn't fix it, we can disable this flag at https://github.com/bazelbuild/bazel/blob/9971445667af255a5e45fadea8b693d6ae91439b/.bazelci/presubmit.yml#L10

**Comment by meteorcloudy** [comment]
Or wait a bit for https://github.com/bazelbuild/bazel/pull/17094 to get merged, which probably upgrades the upb version.

**Pull request closed without merging**